laurent22/joplin
# Desktop: Fixes #11405: Reduce application size by removing unnecessary Rust files

Fixes https://github.com/laurent22/joplin/issues/11405

# Summary

After the inclusion of the `onenote-converter` package to the project, the release build had a 100MB+ increase in size.

| Format  | 3.2.1| 3.2.3 |
|--------|--------|--------|
| AppImage | 238 MB | 369 MB |
| dmg | 232 MB | 367 MB |
| exe | 340 MB | 527 MB |


This PR addresses that by removing the `target` folder used by Rust to store build and was being imported to the application bundle.

After this PR:

| Format  | 3.2.1| 3.2.3 | This PR | 
|--------|--------|--------|--------|
| AppImage | 238 MB | 369 MB | 270 MB|

## Further work:

The biggest contributors to the size are:
![image](https://github.com/user-attachments/assets/bd35be87-81d8-499d-ae40-23b5cb67c5b0)

It seems like the packages are duplicated, for example `@joplin/renderer` requires `mermaid`, but `@joplin/lib` by requiring `@joplin/renderer` also has `mermaid` inside of it.

Not sure how much work, but I think we might be able to cut down some MBs with a way to remove duplication when they happen. (maybe this already happen on the executable and it is only visible when the `asar` is unpacked??)


# Testing

1. Run `git clean -dfx` on project root
2. Run `yarn`
3. Run `cd packages/onenote-converter && IS_CONTINUOUS_INTEGRATION=1 yarn build`
4. Run `cd ../app-desktop`
5. Run `yarn dist`
6. Run `ls -lah dist/`
7. Check size of the executable
8. Run `./dist/Joplin-3.2.3.AppImage` and make sure it is working
